### DISCUSSION TITLE
Customize the ObjectMapper used by DgsQueryExecutor in tests

### DISCUSSION THREAD

**Discussion by nimaaj93** [Netflix/dgs-framework#497] [Q&A]
I have a custom scalar type that is working fine on DGS queries and mutations. However, since I am using the same types to map responses in my tests, I am not getting correct values for these custom types.

I looked into the default `DgsQueryExecutor` injected to my tests and customizing `ObjectMapper` doesnt seem to be possible.

Any way I could provide my custom ObjectMapper for this? 
Or other ways of solving this issue?

**Comment by seanf** [comment]
I have the same problem, trying to deserialise (Default)`Connection` objects using the Relay Cursor Connections feature for pagination.  I want to register `GuavaModule` with the `ObjectMapper`, because `DefaultConnection` uses `ImmutableList` from Guava.

**Comment by Ramblurr** [comment]
Yup, I'm also having the same problem. 

The main goal is to validate a complex query result, as things stands it's very un-ergonomic to do this sort of validation with DGS.

By being able to specify our own objectmapper this would become easier.

**Comment by srinivasankavitha** [comment]
We have a mechanism to set up a custom object mapper for the DGS as a bean: https://netflix.github.io/dgs/advanced/custom-object-mapper/. However, this won't be injected into the DGSQueryExecutor. It's used by the DgsRestController, so you'll need to set up a configuration class to override this bean in your test set up.

I think it does make sense for the DgsQueryExecutor to also have a customizable mapper in this case. Will look into adding that.

**Reply by srinivasankavitha** [reply]
https://github.com/Netflix/dgs-framework/issues/1021

**Reply by seanf** [reply]
Thanks @srinivasankavitha . 

To be clear, without #1021 we need to write our own version of `DefaultDgsQueryExecutor`, which will inject the custom `ObjectMapper` instead of using `com.netflix.graphql.dgs.internal.BaseDgsQueryExecutor#getObjectMapper`?

**Comment by unkali** [comment]
I am facing the same issue in my tests with the DefaultConnection class that uses Guava's Immutable List.
 @srinivasankavitha Is this being considered in the upcoming releases? I will be happy to contribute if this is not in the works yet.
@seanf @nimaaj93 Did you guys find a way to make it work currently? 
I am getting the response as a map for now and then using my custom object mapper to parse it as below:
```
Object response = dgsQueryExecutor.executeAndExtractJsonPath(query, jsonPath);
Connection<MyObject> resp = myCustomobjectMapper.convertValue(response, new TypeReference<DefaultConnection<MyObject>>() {});
```

**Reply by srinivasankavitha** [reply]
Hi @unkali. Thanks, we would welcome PR contributions. We have not yet started working on this, as we have been focused on getting a release with Spring Boot 3 support. Feel free to open a PR and we can incorporate in a future release.
